tasks/skylab/PairedTagUtils.wdl

```diff
@@ -31,36 +24,105 @@ task PairedTagDemultiplex {
         mem_size: "(optional) the amount of memory (MiB) to provision for this task"
         cpu: "(optional) the number of cpus to provision for this task"
         disk_size: "(optional) the amount of disk space (GiB) to provision for this task"
-        preemptible: "(optional) if non-zero, request a pre-emptible instance and allow for this number of preemptions before running the task on a non preemptible machine"
+        preemptible: "(optional) if non-zero, request a pre-emptible instance and allow for this number of preemptions before running the task on a non preemptible machine"        
     }
-
     command <<<
-
-        set -e
-        echo ~{read1_fastq}
-        echo ~{barcodes_fastq}
-        echo ~{read3_fastq}
-        echo Renaming files
+      set -e
+      ## Need to gunzip the r1_fastq
+      zcat ~{barcodes_fastq} | head -n2 > r2.fastq
+      FASTQ=r2.fastq
+      echo 'this is the fastq:' $FASTQ
+      R2=$(awk 'NR==2' $FASTQ)
+      COUNT=$(echo ${#R2})
+      echo 'this is the read:' $R2
+      echo 'this is the barcode count:' $COUNT
+
+      if [[ $COUNT == 27 && ~{preindex} == "false" ]]
+        then
+        pass="true"
```

Comment:
I assume this is just setting the default? You could move this outside the conditionals so that you can set the default once instead of setting it in each condition

Reply:
moved the pass="true" out